File: easy/maxDepth.java
/**
 *  0ms, faster than 100% of java sols
 *  38.9mb, less than 53.14% of java sols
 */

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if(root == null)
            return 0;
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        return max(left, right) + 1;
    }
    
    public int max(int a, int b){
        if(a > b)
            return a;
        return b;
    }
}

File: easy/maxDepth.c
/**
 *  4ms, faster than 92.83% of C sols
 *  8MB, less than 63.76% of C sols
 *
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


int maxDepth(struct TreeNode* root){
    if(root == NULL)
        return 0;
    return max(maxDepth(root->left), maxDepth(root->right)) + 1;
}

int max(int a, int b){
    if(a > b)
        return a;
    return b;
}
